= module Net::HTTPExceptions
HTTP 例外クラスです。

実際にはこれを include した以下のサブクラスの
例外が発生します。

  * [[c:Net::HTTPError]]
  * [[c:Net::HTTPRetriableError]]
  * [[c:Net::HTTPServerException]]
  * [[c:Net::HTTPFatalError]]

また、例外を発生させるためには [[m:Net::HTTPResponse#value]] を
呼ぶ必要があります。

== Instance Methods
--- response -> Net::HTTPResponse

例外の原因となったレスポンスオブジェクトを返します。

#@samplecode 例
require 'net/http'

uri = "http://www.example.com/invalid.html"
response = Net::HTTP.get_response(URI.parse(uri))
begin
  response.value
rescue => e
  e.response # => #<Net::HTTPNotFound 404 Not Found readbody=true>
end
#@end

= class Net::HTTPError < Net::ProtocolError
include Net::HTTPExceptions

HTTP ステータスコード 1xx を受け取ったという例外です。
または、ステータスコードが未知のものである場合も
これに対応します。

= class Net::HTTPRetriableError < Net::ProtoRetriableError
include Net::HTTPExceptions

HTTP ステータスコード 3xx を受け取ったという例外です。

リソースが移動したなどの理由により、リクエストを完了させるには更な
るアクションが必要になります。

= class Net::HTTPServerException < Net::ProtoServerError
#@since 2.6.0
alias Net::HTTPClientException
#@end
include Net::HTTPExceptions

HTTP ステータスコード 4xx を受け取ったという例外です。

クライアントのリクエストに誤りがあるか、サーバにリクエストを拒否さ
れた(認証が必要、リソースが存在しないなどで)ことを示します。

[[c:Net::HTTPServerException]] は Ruby 2.6 から deprecated になりました。
#@since 2.6.0
[[c:Net::HTTPClientException]] を使用してください。
#@end

= class Net::HTTPFatalError < Net::ProtoFatalError
include Net::HTTPExceptions

HTTP ステータスコード 5xx を受け取ったという例外です。

サーバがエラーを起こしているなど、サーバ側で処理を完了することがで
きないことを表しています。

= class Net::HTTPBadResponse < StandardError
HTTP のレスポンスが不正であった場合に発生する例外です。

= class Net::HTTPHeaderSyntaxError < StandardError
HTTP ヘッダの内容が不正である場合に発生する例外です。
